### PR TITLE
ultrahonk finally

### PR DESCRIPTION
**Pull request by signorecello**
This is a working version for Noir 1.0.0-beta.0 using Barretenberg v0.63.0

Uses ultrahonk which is currently slightly broken on the verifier contract generation on the bb side. Commenting out the test until we figure this out (interface won't change)

### CODE REVIEW

**Comment by signorecello** [comment]
cc @olehmisar thanks for all your work with the hardhat plugin!

**Review by critesjosh — CHANGES_REQUESTED** [review]
When going through the vite-hardhat README I am seeing this when running bunx hardhat node . when I resolve this i get other errors about missing packages

The README needs to updated

Screenshot from 2024-12-18 10-37-51

**Comment by critesjosh** [comment]
Should we note the tradeoffs between ultraplonk and ultrahonk in the README? Or are we planning to drop UP altogether so should just forget about it?

**Comment by signorecello** [comment]
We're looking into dropping UP but let's see, depends if we're gonna optimize UH verifier contract or not... still up for debate I guess

**Review by critesjosh — APPROVED** [review]
Would be good to make it more obvious that UH on-chain verification isn't working, but otherwise looks good!

**Comment by Savio-Sou** [comment]
Re UltraPlonk and UltraHonk:
- The Barretenberg team plans to audit UH only but not UP, best not to recommend UP
- The Barretenberg team plans to optimize the UH Solidity verifier further, e.g. https://github.com/AztecProtocol/aztec-packages/pull/8690

**Comment by Savio-Sou** [comment]
This PR resolves https://github.com/AztecProtocol/noir-starter/issues/97

**Comment by critesjosh** [comment]
Makes sense, thanks @Savio-Sou. We were discussing also switching back to UP because the UH verifier is broken, so we could have a fully functioning example. 

What do you think, should we merge it with UH, make a note that the UH verifier is being worked on and will be updated soon?

**Comment by Savio-Sou** [comment]
Merging with UH + note sounds reasonable.

Which / Do we have an Issue for UH's broken verifier? We should prioritize fixing it in parallel.

**Comment by signorecello** [comment]
Just because I was already working on it, I'm pushing a refactor that allows for both UltraPlonk and UltraHonk. It also tests both, but skips the verifier contract for UltraHonk.

**Comment by signorecello** [comment]
> Which / Do we have an Issue for UH's broken verifier? We should prioritize fixing it asap in parallel.

We don't, I have a pretty good idea of what could be missing so I'm looking into it so I can provide more context to the team

File: vite-hardhat/README.md [hunk 1]
```diff
@@ -49,11 +49,17 @@ Want to get started in a pinch? Start your project in a free Github Codespace!
...
-bun test
+bun run test
 ```
 
 This test shows the basic usage of Noir in a TypeScript Node.js environment. It also starts its own network and deploys the verifier contract.
 
+If you want to test only `UltraHonk`, you can run:
+
+```bash
+bun run test UltraHonk
```
Comment (critesjosh) [inline vite-hardhat/README.md:60]: Seeing this error when I run this line. Other than that, looks good.

Screenshot from 2024-12-19 12-25-56

Comment (signorecello) [reply vite-hardhat/README.md]: fixed :)

**Pull request merged**